src/modules/block.js

```diff
@@ -13,17 +13,20 @@ module.exports = bot => {
       msg.channel.createMessage(`Blocked <@${userId}> (id ${userId}) from modmail`);
     }
 
-    let logText = `**Blocked:** ${thread.user_name} (${thread.user_id}) was blocked.`;
+    let logText = `**Blocked:** `;
 
     if (! thread && args.length > 0) {
       // User mention/id as argument
-      const userId = utils.getUserMention(args.join(' '));
+      const userId = utils.getUserMention(args[0]);
       if (! userId) return;
 
+      const user = bot.users.get(userId);
       const reason = args.slice(1).join(' ').trim();
 
+      logText += `${(user ? `${user.username}#${user.discriminator}` : 'Unable to fetch username')} (${userId}) was blocked`;
```

Comment:
Probably better if we fetched from REST here for people who have left the guild - would also eliminate IDs which aren't bound to a user

Reply:
Well I figured that too, however that wasn't how it worked in the function to block users, so I just did the same as I wasn't sure if there was a reason you didn't use REST there

Reply:
Yeah change it to use REST as well in the block function - since we're making the function async, we'll also need to await on lines which call it, this way we can handle errors properly

Reply:
Yep, no worries. Will do that now